app/benchmarks/gas_estimator_test.go

```diff
@@ -0,0 +1,126 @@
+	infos := queryAccountInfo(testApp, accounts, kr)
+	signer, err := user.NewSigner(kr, encfg.TxConfig, testutil.ChainID, appconsts.LatestVersion, user.NewAccount(accounts[0], infos[0].AccountNum, infos[0].Sequence))
+	require.NoError(b, err)
+
+	benchmarks := []struct {
+		name                 string
+		pfbSize              int
+		numberOfTransactions int
+	}{
+		{
+			name:                 "10kb PFBs filling 120% of a block",
+			pfbSize:              10_000,
+			numberOfTransactions: (appconsts.DefaultMaxBytes * 120 / 100) / 10_000,
+		},
+		{
+			name:                 "100kb PFBs filling 120% of a block",
+			pfbSize:              100_000,
+			numberOfTransactions: (appconsts.DefaultMaxBytes * 120 / 100) / 100_000,
+		},
+		{
+			name:                 "1mb PFBs filling 120% of a block",
+			pfbSize:              1_000_000,
+			numberOfTransactions: (appconsts.DefaultMaxBytes * 120 / 100) / 1_000_000,
+		},
+		{
+			name:                 "2mb PFBs filling 120% of a block",
+			pfbSize:              2_000_000,
+			numberOfTransactions: (appconsts.DefaultMaxBytes * 120 / 100) / 2_000_000,
+		},
+	}
+
+	for _, bench := range benchmarks {
+		b.Run(bench.name, func(b *testing.B) {
+			txs := generateRandomPFBs(b, signer, accounts[0], bench.numberOfTransactions, bench.pfbSize)
+			mempool := benchMempool{txs: txs}
+			gasEstimationServer := gasestimation.NewGasEstimatorServer(
+				mempool,
+				encfg.TxConfig.TxDecoder(),
+				func(txBytes []byte) (sdk.GasInfo, *sdk.Result, error) { return sdk.GasInfo{}, nil, nil },
+			)
+			for i := 0; i < b.N; i++ {
+				_, err := gasEstimationServer.EstimateGasPrice(context.Background(), &gasestimation.EstimateGasPriceRequest{})
+				require.NoError(b, err)
+			}
+		})
+	}
+}
+
+func generateRandomPFBs(b *testing.B, signer *user.Signer, account string, numberOfTransactions int, blobSize int) []types.Tx {
+	rand := tmrand.NewRand()
+	gasLimit := blobtypes.DefaultEstimateGas([]uint32{uint32(blobSize)})
+	blobs := blobfactory.ManyBlobs(rand, []share.Namespace{share.RandomBlobNamespace()}, []int{blobSize})
+	txs := make([]types.Tx, numberOfTransactions)
+	bTxFee := rand.Uint64() % 10000
+	for i := 0; i < numberOfTransactions; i++ {
+		bTx, _, err := signer.CreatePayForBlobs(account, blobs,
+			user.SetFee(bTxFee),
+			user.SetGasLimit(gasLimit))
+		require.NoError(b, err)
+		txs[i] = bTx
+	}
+	return txs
+}
+
+var _ client.MempoolClient = &benchMempool{}
+
+type benchMempool struct {
+	txs []types.Tx
+}
+
+func (b benchMempool) UnconfirmedTxs(ctx context.Context, limit *int) (*rpctypes.ResultUnconfirmedTxs, error) {
+	return &rpctypes.ResultUnconfirmedTxs{Txs: b.txs}, nil
+}
+
+func (b benchMempool) NumUnconfirmedTxs(ctx context.Context) (*rpctypes.ResultUnconfirmedTxs, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (b benchMempool) CheckTx(ctx context.Context, tx types.Tx) (*rpctypes.ResultCheckTx, error) {
+	//TODO implement me
+	panic("implement me")
```

Comment:
can this return empty values instead of panicing?

Reply:
f084da0bd97346b6278b7cb784cd1bf444065f06